cmd/wallet-sdk-gomobile/docs/packages.md

```diff
@@ -0,0 +1,400 @@
+generates the bindings for the two platforms. Due to these differences (and the limitations of `gomobile`),
+it's difficult to pick names that look perfect across both platforms and the Go source code.
+Thus, the various names used in the Go files were chosen based on achieving a reasonable compromise between all
+three versions.
+
+
+## Android
+
+* `dev.trustbloc.wallet.sdk.api`
+* `dev.trustbloc.wallet.sdk.credential`
+* `dev.trustbloc.wallet.sdk.credentialsigner`
+* `dev.trustbloc.wallet.sdk.credentialverifier`
+* `dev.trustbloc.wallet.sdk.didcreator`
+* `dev.trustbloc.wallet.sdk.didresolver`
+* `dev.trustbloc.wallet.sdk.linkeddomains`
+* `dev.trustbloc.wallet.sdk.localkms`
+* `dev.trustbloc.wallet.sdk.memstorage`
+* `dev.trustbloc.wallet.sdk.openid4ci`
+* `dev.trustbloc.wallet.sdk.openid4vp`
+* `dev.trustbloc.wallet.sdk.presentationexchange`
+* `dev.trustbloc.wallet.sdk.vdr`
+* `dev.trustbloc.wallet.sdk.wallethandle`
+
+## iOS
+
+Everything is under a single package named `Walletsdk`.
+
+Interfaces (protocols), types, and functions are all prefixed according to their corresponding Go package names.
+Note that this behaviour is enforced by the `gomobile` tool and cannot currently be changed.
+See [here](https://github.com/golang/go/issues/32573) for more information.
+
+Prefixes:
+
+* `Api`
+* `Credential`
+* `Credentialsigner`
+* `Credentialverifier`
+* `Didcreator`
+* `Didresolver`
+* `Linkeddomains`
+* `Localkms`
+* `Memstorage`
+* `Openid4ci`
+* `Openid4vp`
+* `Presentationexchange`
+* `Vdr`
+* `Wallethandle`
+
+## Package/Module Examples
+
+To better understand how package naming works, it's helpful to look at some examples for each of the packages/modules
+and compare how something in Go gets converted to Java (for Android) and Objective-C (for iOS).
+
+These examples are not meant to be full code examples of how you actually use the API.
+Rather, these examples are just to help you understand the naming patterns, which should help you in figuring out
+the equivalent names for various parts of the API across platforms.
+
+The left-most side in the tables shows Go source code snippets, rather than how they would look when imported in another
+file. The middle and right-most side show how the code looks when importing that same functionality in via the bindings.
+The reason for comparing between Go and the bindings like this is that the Go code is not intended to be imported
+directly - the mobile wrappers exist only to make the Go code compatible with the `gomobile` tool, so seeing how it
+would look imported is not so useful. However, the way you import from the generated mobile bindings is important,
+and so that's why we show that here. The Go source code is really just meant to help demonstrate the naming patterns
+to the reader (as the names used for packages/modules in the bindings are generated automatically by `gomobile` based
+on the Go names).
+
+### API
```

Comment:
In the future, we should consider finding a better way to organize the stuff in the `api` folder. I don't think it's very clear to a caller why certain things are in here, or why it's named `api`. Right now it's basically a "common" type of folder.